doc/api/v8.md

```diff
@@ -1057,15 +1057,47 @@ Returns true if the Node.js instance is run to build a snapshot.
 added: v19.6.0
 -->
 
-This API collects GC data in current thread.
+### `GCProfile.GC_PROFILER_MODE`
 
-### `new v8.GCProfiler()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `GCProfile.GC_PROFILER_MODE.TIME`: Only the GC total time will be collected.
+* `GCProfile.GC_PROFILER_MODE.ALL`: the GC total time and heap statistic will
+  be collected.
+
+Define which data the `GCProfile` will collect.
+
+### `GCProfile.GC_PROFILER_MODE`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `GCProfile.GC_PROFILER_FORMAT.NONE`: return `undefined`.
+* `GCProfile.GC_PROFILER_FORMAT.OBJECT`: return an `object`.
+* `GCProfile.GC_PROFILER_FORMAT.STRING`: return a `string`.
+
+Define the return value in `stop` API.
+
+### `new v8.GCProfiler([options])`
 
 <!-- YAML
 added: v19.6.0
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/46733
+    description: support GC mode and format.
 -->
 
-Create a new instance of the `v8.GCProfiler` class.
+* `options` {Object}
+  * `mode` {number} which data will be collected, defined by
+    `GCProfiler.GC_PROFILER_MODE`.
+    **Default:** `GCProfiler.GC_PROFILER_MODE.ALL`.
+
+Create a new instance of the `v8.GCProfiler` class, This instance collects GC
+data in current thread.
```

Comment:
I think `in current thread` is somewhat ambiguous when the GC itself can involve multiple threads. Perhaps

```suggestion
statistics in the current Node.js instance.
```